Headers/Foundation/NSURLSession.h

```diff
@@ -430,6 +437,37 @@ didReceiveChallenge: (NSURLAuthenticationChallenge*)challenge
 
 @end
 
+@protocol NSURLSessionDownloadDelegate <NSURLSessionTaskDelegate>
+
+/* Sent when a download task that has completed a download.  The delegate should 
+ * copy or move the file at the given location to a new location as it will be 
+ * removed when the delegate message returns. URLSession:task:didCompleteWithError: will
+ * still be called.
+ */
+- (void)         URLSession: (NSURLSession *)session
+               downloadTask: (NSURLSessionDownloadTask *)downloadTask
+  didFinishDownloadingToURL: (NSURL *)location;
+
+@optional
```

Comment:
For optional parts we normally use something like this:

#if GS_PROTOCOLS_HAVE_OPTIONAL
@optional
#else
@end
@interface NSObject (NSURLSessionDownloadDelegate)
#endif // GS_PROTOCOLS_HAVE_OPTIONAL

I am not sure whether this is still needed for any compiler that we still support. GNUstep is very conservative, we try to keep stuff running even for very old systems.

Reply:
GCC 4.6 added the ´´´@optional´´´ directive.

Reply:
And NSURLSession uses blocks so it will only compile with clang. So this is not needed.

Reply:
These are good enough reasons to use @optional directly.